src/util.c

```diff
@@ -2581,54 +2624,58 @@ nats_freeMetadata(natsMetadata *md)
 natsStatus nats_formatStringArray(char **out, const char **strings, int count)
 {
     natsStatus s = NATS_OK;
-    natsBuffer buf;
-    int len = 0;
-    int  i;
+    natsBuffer buf = { 0 };
 
-    len++; // For the '['
-    for (i = 0; i < count; i++)
+    s = natsBuf_Init(&buf, 128);
+    IFOK(s, nats_marshalStringArray(&buf, false, NULL, strings, count));
+    IFOK(s, natsBuf_AppendByte(&buf, '\0'));
+    if (s != NATS_OK)
     {
-        len += 2; // For the quotes
-        if (i > 0)
-            len++; // For the ','
-        if (strings[i] == NULL)
-            len += (int)strlen("(null)");
-        else
-            len += (int)strlen(strings[i]);
+        natsBuf_Cleanup(&buf);
+        return s;
     }
-    len++; // For the ']'
-    len++; // For the '\0'
 
-    s = natsBuf_Init(&buf, len);
+    *out = natsBuf_Data(&buf);
+    return NATS_OK;
+}
+
+// note: does not JSON-escape the strings
+natsStatus
+nats_marshalStringArray(natsBuffer *buf, bool comma, const char *fieldName, const char **values, int len)
+{
+    natsStatus s = NATS_OK;
+    int  i;
+    const char *sep = (comma ? "," : "");
+
+    if (!nats_IsStringEmpty(fieldName))
+    {
+        s = natsBuf_Append(buf, sep, -1);
```

Comment:
I checked the code of `natsBuf_Append` and I guess that works with `sep` being `""`, but we could "optimize" `natsBuf_Append` and return early when we evaluate `dataLen` and `dataLen == 0`.